src/cpon.rs

```diff
@@ -952,5 +959,12 @@ mod test
         //let cpon2 = rv.to_cpon_string().unwrap();
         //assert_eq!(cpon1, cpon2);
     }
+    #[test]
+    fn test_read_too_long_numbers() {
+        // read very long decimal without overflow error, value is capped
+        assert!(RpcValue::from_cpon("123456789012345678901234567890123456789012345678901234567890").unwrap().is_int());
```

Comment:
In libshv-js, this returns INT_MAX_SAFE_INTEGER. The capped value should be tested also. It's not immediately clear to whether the code (with the checked_mul) actually caps it at intmax (or something like that), or if it just refuses to multiply when the overflow happens.